### ISSUE TITLE
MacOS: Context leak detected, msgtracer returned -1

### ISSUE THREAD

**Issue by altunenes** [thewh1teagle/sherpa-rs#23]
Hello! Thank you for this work again! I tried to run the example exactly https://github.com/thewh1teagle/sherpa-rs/blob/main/examples/diarize.rs here with:
"
    let segment_model_path = "model.onnx"; //latest seg. model from sherpa...
    let embedding_model_path = "wespeaker_en_voxceleb_CAM++.onnx";
    let wav_path = "normalized_audio.wav";"

Context leak detected, msgtracer returned -1
Context leak detected, msgtracer returned -1
Context leak detected, msgtracer returned -1
🗣️ Diarizing... 0% 🎯
Context leak detected, msgtracer returned -1
Context leak detected, msgtracer returned -1
🗣️ Diarizing... 0% 🎯
🗣️ Diarizing... 0% 🎯
🗣️ Diarizing... 1% 🎯
🗣️ Diarizing... 1% 🎯
🗣️ Diarizing... 1% 🎯
🗣️ Diarizing... 1% 🎯
🗣️ Diarizing... 2% 🎯
🗣️ Diarizing... 2% 🎯

The code works without any problems. However, I wanted to bring this warning message to your attention.

**Comment by thewh1teagle** [comment]
@altunenes 
I can't reproduce on Windows 11.
Does it happens with the latest version of sherpa-rs?

**Comment by altunenes** [comment]
yes, it still persists. Also, I see this on the pyannote too and maybe whisper-rs related thing. 

I've never seen it on Windows either. It's like a “warning” message that only appears on macos and it's very difficult to understand its origin. 

note: Since it doesn't prevent the code from working and doesn't cause a major “leak” (at least as far as I can see), I didn't care too much, but I wanted to let you know anyway. :-)

**Comment by feelingsonice** [comment]
Same here on MacOs, except it happens when running the `tts` example using Piper English (commands from the first comment). 

Something notable is that first time running gave me "Context leak detected, msgtracer returned -1" followed by "Segmentation fault". Ran it again without doing anything and it produced the proper `audio.wave` file instead of the "segmentation fault".

**Comment by thewh1teagle** [comment]
> Something notable is that first time running gave me "Context leak detected, msgtracer returned -1" followed by "Segmentation fault". Ran it again without doing anything and it produced the proper `audio.wave` file instead of the "segmentation fault".

Try to set the provider to `cpu` by `--provider cpu`

**Comment by feelingsonice** [comment]
```text
$ cargo run --example tts --features="tts" -- --provider cpu --text 'liliana, the most beautiful and lovely assistant of our team!' --output audio.wav --tokens "vits-piper-en_US-amy-low/tokens.txt" --model "vits-piper-en_US-amy-low/en_US-amy-low.onnx" --data-dir "vits-piper-en_US-amy-low/espeak-ng-data"

    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.10s
     Running `target/debug/examples/tts --provider cpu --text 'liliana, the most beautiful and lovely assistant of our team'\!'' --output audio.wav --tokens vits-piper-en_US-amy-low/tokens.txt --model vits-piper-en_US-amy-low/en_US-amy-low.onnx --data-dir vits-piper-en_US-amy-low/espeak-ng-data`
Context leak detected, msgtracer returned -1
Context leak detected, msgtracer returned -1
Context leak detected, msgtracer returned -1
Context leak detected, msgtracer returned -1
Context leak detected, msgtracer returned -1
Context leak detected, msgtracer returned -1
Created audio.wav
```
Same thing as the second run unless I'm doing something wrong?

**Comment by thewh1teagle** [comment]
> Same thing as the second run unless I'm doing something wrong?

the provider argument wasn't used in the example so so it still used `coreml`. I fixed it now so you can `git pull` and run again.

**Comment by feelingsonice** [comment]
Yeah that worked :)

**Comment by thewh1teagle** [comment]
> Yeah that worked :)

Great! I also noticed that it works faster on CPU with macOS m1. weird.

**Comment by feelingsonice** [comment]
Yeah I noticed it too. Worked faster for the `whisper` example too. Maybe cuz the models are small and offloading to the GPU introduces a lot overhead?

**Comment by thewh1teagle** [comment]
> Yeah I noticed it too. Worked faster for the `whisper` example too. Maybe cuz the models are small and offloading to the GPU introduces a lot overhead?

I think that it's related to onnxruntime operators. the platform claims to 'support' many backends. in the reality It didn't worked for me fast in them (Eg. coreml, directml)

**Comment by altunenes** [comment]
I do not get any warning msg on Windows 11
both CUDA & CPU
